### DISCUSSION TITLE
Most features not working

### DISCUSSION THREAD

**Discussion by UltraBlackLinux** [andyholmes/valent#381] [Help]
### Code of Conduct

- [X] I will prove that I'm an awesome person by using my manners
- [X] I agree that I am only entitled to [The Four Essential Freedoms](https://www.gnu.org/philosophy/free-sw.html#four-freedoms)
- [X] I understand this project is in alpha and bugs are expected

### Current Behavior

Somehow almost every feature is not working at all.

The remote input, while looking complete, did not do anything (wayland issue?), the presentation remote only worked when it came to uploading the file and the media remote did receive the media data, but could not send any.

The "browse files" button does not seem to do anything once configured. Nothing is mounted or accessible.

### Expected Behavior

I know that not everything is 100% done yet, but I am a little confused regardless. The only features that I got to work were ping and ring.

Is this a problem with my setup? I explicitly only enabled kdeconnect via firewalld on both my phone and my desktop.

Now what I find confusing is that the coverage report states almost 100% for all of these features and yet most of them are completely unusable.

### Steps to Reproduce

I guess I described that all above

### Desktop

GNOME Shell

### Other Desktop

Phosh though. only somewhat comparable to the Proper gnome shell.

### Operating System

DanctNIX (Arch)

### Installed from

Distribution Package

### Version

1.0.0.alpha

### Devices

Valent (Desktop Linux), Valent (Phosh)

### Logs

```shell
Could sadly not get any as the process detaches itself from any terminal immediately and there is seemingly no way to circumvent this.
```

**Comment by andyholmes** [comment]
I'm guessing you installed this through AUR? I haven't done that in a long time, so I'm not sure it pulls in all the required dependencies.

Are there really no logs in `journalctl` or otherwise?

**Comment by UltraBlackLinux** [comment]
Yeah I did install that from the AUR.

I inspected the journal but I couldn't find anything there.

I think it automatically pulled the correct dependencies, otherwise it wouldn't have build, would it? The build command in the PKGBUILD is also the same as listed in this repository.

Could you add a CLI flag to prevent the process from detaching?

**Comment by symbioquine** [comment]
I also installed from AUR via the `valent-git` package. At risk of confusing this issue or making it track too many things, here's what seems to be working for me;

* Battery & Signal Strength icons (Signal strength appears to only show cell signal)
* Messaging
* Send Files (both directions)
  - From phone to computer the file appeared in my downloads folder
  - From computer to phone, the file appeared in my primary storage location
* Ping
* Take Photo
  - The file (slightly surprisingly) appeared in my home folder
* Media Remote
  - Play/pause works
  - Volume control works
  - Scrub chapter position works in NewPipe, Audible, and VLC
  - Forward/backward by chapter/track works with Audible and VLC, but not NewPipe
  - Forward/backward by 10s didn't work with any of the tested players
  - I didn't test the repeat/shuffle options
* Ring

Things that don't seem to be working at all;

* Browse Files
  `valent-sftp-plugin:  WARNING: sshadd_cb(): Failed to add host key: Child process exited with code 2`
* Send clipboard (from phone)
  - Nothing seems to happen
* Remote Input (both directions)
  - From phone to computer: `valent-xdp-input:  WARNING: on_session_created(): GDBus.Error:org.freedesktop.DBus.Error.UnknownMethod: No such interface “org.freedesktop.portal.RemoteDesktop” on object at path /org/freedesktop/portal/desktop`
* Battery status for desktop from phone (shows -1% - presumably because my desktop doesn't have a battery)

I didn't test the "Presentation Remote"/"Slideshow Remote" features

Based on my testing, it seems like more than half the features are working. This is a great tool, thanks for working on it!

**Comment by andyholmes** [comment]
@UltraBlackLinux 

> I inspected the journal but I couldn't find anything there.

Unfortunately without some logs, it's pretty hard to troubleshoot. Valent is actually pretty noisy in it's current state, so there should be something.

> I think it automatically pulled the correct dependencies, otherwise it wouldn't have build, would it?

There are both build and runtime dependencies; for example it seems Arch does not come with the RemoteDesktop portal installed, which is the only interface with a default plugin in Valent.

> Could you add a CLI flag to prevent the process from detaching?

Unless you're starting Valent with something like `gapplication launch` or D-Bus activation, it should just run in the terminal:

![Screenshot from 2023-02-26 12-02-10](https://user-images.githubusercontent.com/1265208/221434124-974dbc30-d188-4d13-8484-2acd6fe4534e.png)

**Comment by andyholmes** [comment]
@symbioquine 

> At risk of confusing this issue or making it track too many things, here's what seems to be working for me

Thanks, more data points from the same distro is helpful!

> (Signal strength appears to only show cell signal)

That's correct behaviour, other signals are not (really) supported by the protocol yet.

> Take Photo: The file (slightly surprisingly) appeared in my home folder

I think it should go into `$HOME/Pictures`, this might be a bug or a problem with XDG variables. You could open an issue for this if you wanted.

> MediaRemote: Forward/backward by 10s didn't work with any of the tested players

This is sort of expected. KDE Connect technically supports all the media controls, but not every client implements them all or all of the same ones (it's a long explanation for this comment). You could open an issue for this also if you like, although it may not be fixable at the moment :)

> Browse Files
> `valent-sftp-plugin:  WARNING: sshadd_cb(): Failed to add host key: Child process exited with code 2`

This sounds like there's a problem with the ssh-agent, but further research is required. We could likely have a better error message here. Also good bug material.

Note however, that the insecure cipher problem that affects clients connecting to KDE Connect Android still applies here; you need something like this in your `~/.ssh/config` if you expect it to connect (after the ssh-agent problem is fixed):

```
Host 192.168.88.*
  HostKeyAlgorithms +ssh-rsa
```

> Send clipboard (from phone)

Double check the settings in Valent, because I believe this is off by default. Additionally, if you are on Wayland, you will either need the GNOME Shell extension for the clipboard to work or write a plugin for your desktop that does the same.

> Remote Input (both directions) 

With the error report there, this confirms to me that there are run-time dependencies that may be unexpectedly missing. Usually if you have Flatpak installed, you'd have that portal available; maybe that's something that can be added as a runt-time dep or recommended package in AUR?

> Battery status for desktop from phone (shows -1% - presumably because my desktop doesn't have a battery)

That's a bug, for sure.

**Comment by UltraBlackLinux** [comment]
> Unfortunately without some logs, it's pretty hard to troubleshoot. Valent is actually pretty noisy in it's current state, so there should be something.

There is no output in journalctl. I just rechecked, and there is nothing at all.

> There are both build and runtime dependencies; for example it seems Arch does not come with the RemoteDesktop portal installed, which is the only interface with a default plugin in Valent.

Ah I see.
 
> Unless you're starting Valent with something like gapplication launch or D-Bus activation, it should just run in the terminal.

No, it just detaches. I can't seem to control that in any way. The only output I got was the following
```
GLib-GIO:    DEBUG: Using cross-namespace EXTERNAL authentication (this will deadlock if server is GDBus < 2.73.3)
GLib-GIO:    DEBUG: _g_io_module_get_default: Found default implementation gvfs (GDaemonVfs) for ‘gio-vfs’
GLib-GIO:    DEBUG: _g_io_module_get_default: Found default implementation gnutls (GTlsBackendGnutls) for ‘gio-tls-backend’
```

**Comment by andyholmes** [comment]
> No, it just detaches. I can't seem to control that in any way. The only output I got was the following

That sounds like `valent` is still running when you execute the program. Since it's a single-instance D-Bus application, the `valent` executable returns if it can't own the service name on D-Bus.

Can you try running `killall -15 valent; valent` from the terminal?

**Comment by isshin1** [comment]
hope im allowed to share my details here as well, really happy to see a kdeconnect app for non kde/gnome DEs.
using kdeconnect playstore app on Pixel 7 A13,  valent-git from AUR, arch linux


### whats working for me 
- battery 
- network
- file transfer 
- ping
- media remote
- ping
- ring
- notifications 

### whats not working
-  browsing files 
> WARNING: mount_cb(): Error mounting: Connection failed


- clipboard
this plugin imo is most imp feature, im continously getting this 
>  valent_clipboard_plugin_handle_clipboard_connect(): expected "content" field holding a string
>  valent-clipboard-plugin:  WARNING: valent_clipboard_read_text_cb(): No compatible transfer format found

- camera 
Everytime I click the Take photo, camera app is opened in phone with an animation ( even if its opened already ) 
and it doesnt take any pics, and nothing in logs either. Only diff I can see is that when camera is opened using valent , I cant see the icon which let you access recent photos, so im assuming somehow it doesnt have access to location.

normal camera app
![image](https://user-images.githubusercontent.com/36916613/222048506-668f6010-2f25-4e75-8e72-357a844a20e8.png)

camera app when opened using valent 
![image](https://user-images.githubusercontent.com/36916613/222048536-7bb4f0ba-8fb1-4532-9044-ee088eb1cc16.png)

also seems like a camera instance is opened everytime i press take photo, had to swipe back multiple times to return back to kdeconnect app.

**Comment by andyholmes** [comment]
> hope im allowed to share my details here as well, really happy to see a kdeconnect app for non kde/gnome DEs.

Sure thing! I'm going to convert this issue into a discussion though, since it's pretty all over the place :slightly_smiling_face: 

> clipboard
> this plugin imo is most imp feature, im continously getting this

Are you using Wayland? If so, it is unlikely the clipboard feature will work, unless you're using GNOME Shell and can install the extension.

Otherwise, you should open a new issue with some details about what you're trying to copy and so on :)

> browsing files
> `WARNING: mount_cb(): Error mounting: Connection failed`

This sounds like it's probably the same bug as [GSConnect#1203](https://github.com/GSConnect/gnome-shell-extension-gsconnect/issues/1203). See [my comment above](https://github.com/andyholmes/valent/issues/372#issuecomment-1445458332) for overriding insecure ciphers.

> Everytime I click the Take photo, camera app is opened in phone with an animation ( even if its opened already )
and it doesnt take any pics, and nothing in logs either.

That's the correct behaviour for the Photo plugin, at least for Android. The feature was intended for "scan my boarding pass" situations, where you press "Take Photo" then when ready press the button.

**Comment by isshin1** [comment]
ye, indeed am on wayland ( sway ) whats the hurdle here tho ? imo everytime a clipboard change is detected, sending it to clipboard should be feasible using wl-clipboard, (atleast for wlroots based compositors ), just a guess.

also about the mounting fs, it works when i launch valent using terminal,  but not directly. not sure whats the cause behind it

**Reply by andyholmes** [reply]
> ye, indeed am on wayland ( sway ) whats the hurdle here tho ? 

X11 exposed the clipboard for any process, without restriction. Wayland has no such guarantees, so it depends on the compositor. If the default clipboard plugin is not working, someone will have to contribute a plugin for your desktop since I'm not familiar with sway or what IPC it supports.

> also about the mounting fs, it works when i launch valent using terminal, but not directly. not sure whats the cause behind it

Valent can't really behave differently depending on how it's started, unless you are starting it in entirely different environments. If the configuration is set as above, SFTP should work or throw another error.

Without more information, I would only be speculating :shrug:
